### PR TITLE
Unit-Testing using contentctl

### PR DESCRIPTION
**Pull request by patel-bhavin**
This PR enables Unit-testing in Github Actions CI using contentctl. 

Additionally added a script to pretty print the pass/failures of detections from summary.yml

image

### CODE REVIEW

File: .github/workflows/unit-testing.yml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+            pip install contentctl
+            git clone --depth=1 --single-branch --branch=master https://github.com/redcanaryco/atomic-red-team.git
+        
+        - name: Run ContentCTL test for changes against develop
+          run: |
+            git pull
+            source .venv/bin/activate
+            git checkout ${GITHUB_REF#refs/heads/}
+            # cat summary.yml
+            # contentctl test --post-test-behavior never_pause mode:changes --mode.target-branch develop
```
Comment (patel-bhavin) [inline .github/workflows/unit-testing.yml]: enable this before merging!

**Comment by patel-bhavin** [comment]
Updated the output for better formatting in summary
image

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -23,29 +21,20 @@ jobs:
         run: |
```
Comment (pyth0n1c) [inline .github/workflows/build.yml]: I don't think this workflow needs jq (anymore) can it be removed entirely?

File: .github/workflows/unit-testing.yml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+            ref: develop
+
+        - uses: actions/setup-python@v5
+          with:
+            python-version: '3.11' #Available versions here - https://github.com/actions/python-versions/releases  easy to change/make a matrix/use pypy
+            architecture: 'x64' # optional x64 or x86. Defaults to x64 if not specified
+
+        - name: Install System Packages and contentctl 
+          run: |
+            sudo apt update -qq
```
Comment (pyth0n1c) [inline .github/workflows/unit-testing.yml]: Now that we are doing the processing with python i think we can remove jq?

**Review by pyth0n1c — DISMISSED** [review]
I believe this looks good. After it is merged, we should do some stress-testing where we open a PR from an external (non splunk/security_content member) repo to ensure that the mode:changes functionality works from external forks as well.

File: .github/workflows/build.yml (hunk 2 above)
Comment (patel-bhavin) [reply .github/workflows/build.yml]: removed!

File: .github/workflows/unit-testing.yml (hunk 3 above)
Comment (patel-bhavin) [reply .github/workflows/unit-testing.yml]: removed

**Comment by patel-bhavin** [comment]
This workflow was also tested on PR from Dean Luxton (Splunker) and it seemed to have worked correctly! You can check the run here : 

https://github.com/splunk/security_content/pull/3024

**Review by pyth0n1c — APPROVED** [review]
Reviewed workflow and results.  Looks great! Approved!

**Pull request merged**